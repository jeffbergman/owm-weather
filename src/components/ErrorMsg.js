import React from 'react';
import { connect } from 'react-redux';

const ErrorMsg = (props) => {
	//just displays an error message if one exists

	//just a dummy error message to see how git works
	
	if (!props.errMsg) return null;

  return (
    <div className="error">{props.errMsg}</div>
  );
}

const mapStateToProps = (state) => {
	return {
  	errMsg: state.error
  };
};

export default connect(mapStateToProps)(ErrorMsg);