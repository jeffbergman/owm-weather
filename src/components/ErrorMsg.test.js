import React from 'react';
import { shallow } from 'enzyme';
import ErrorMsg from './ErrorMsg';

describe('ErrorMsg', () => {
	
	const props = { errMsg: 'city not found' };

	const errorMsg = shallow(<ErrorMsg.WrappedComponent {...props} />);

	it('renders correctly', () => {
		expect('errorMsg').toMatchSnapshot();
	});


	it('displays an error message from props', () => {
		expect(errorMsg.find('.error').text()).toEqual('city not found')
	});

});