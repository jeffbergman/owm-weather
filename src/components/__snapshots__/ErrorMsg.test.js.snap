// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg renders correctly 1`] = `"errorMsg"`;
